app/src/components/PgpSection.tsx

```diff
@@ -0,0 +1,251 @@
+  Text,
+  KeyboardAvoidingView,
+  ScrollView,
+  Alert,
+  Switch
+} from "react-native";
+import { Button } from ".";
+import { useClaimsStore, useClaimValue } from "../context/ClaimsStore";
+import usePgp from "../hooks/usePpg";
+import { AlertTitle, ClaimTypeConstants } from "../constants/common";
+import { pgpLocalStorage } from "../utils/local-storage";
+import {
+  extractPrivateKeyFromContent,
+  formatFingerPrint
+} from "../utils/pgp-utils";
+import { useActionSheet } from "@expo/react-native-action-sheet";
+import QRCode from "react-native-qrcode-svg";
+import colors from "../styles/colors";
+import { TextInput } from "react-native-gesture-handler";
+
+type Props = {
+  emailInput: string;
+  isEmailVerified: boolean | undefined;
+};
+
+const PgpSection: React.FC<Props> = (props) => {
+  const emailClaimValue = useClaimValue(ClaimTypeConstants.EmailCredential);
+  const nameClaimValue = useClaimValue(ClaimTypeConstants.NameCredential);
+  const [isActive, setIsActive] = React.useState(false);
+  const { showActionSheetWithOptions } = useActionSheet();
+  const [publicKey, setPublicKey] = React.useState<string>();
+  const { addClaim } = useClaimsStore();
+
+  const {
+    generateKeyPair,
+    generateKeyPairFromPrivateKey,
+    resendVerificationEmail,
+    importPrivateKeyFileFromDevice
+  } = usePgp();
+
+  const [pgpTitle, setPgpTitle] = React.useState<string | undefined>();
+
+  const loadKeyFromLocalStorage = React.useCallback(async () => {
+    const key = await pgpLocalStorage.get();
+    if (!key) return;
+    setPublicKey(key.publicKey);
+    const fingerPrint = formatFingerPrint(key.fingerPrint);
+    setPgpTitle(fingerPrint);
+  }, [setPublicKey]);
+
+  const extractAndLoadKeyPairFromContent = React.useCallback(
+    async (content: string, email: string) => {
+      const privateKey = extractPrivateKeyFromContent(content);
+      await generateKeyPairFromPrivateKey(privateKey, email);
+      await loadKeyFromLocalStorage();
+    },
+    [generateKeyPairFromPrivateKey, loadKeyFromLocalStorage]
+  );
+
+  const importPrivateKeyFromDevice = React.useCallback(
+    async (email: string) => {
+      try {
+        const content = await importPrivateKeyFileFromDevice();
+        if (!content) return;
+        await extractAndLoadKeyPairFromContent(content, email);
+      } catch (error: any) {
+        Alert.alert(
+          AlertTitle.Error,
+          `Failed to extract the Private Key from file \n> ${
+            error?.message ?? "unknown error"
+          }`
+        );
+      }
+    },
+    [extractAndLoadKeyPairFromContent]
+  );
+  const generateAndPublishNewPgpKey = React.useCallback(
+    async (name: string, email: string) => {
+      await generateKeyPair(name, email);
+      await addClaim(ClaimTypeConstants.EmailCredential, { email }, [], false);
+      await loadKeyFromLocalStorage();
+    },
+    [generateKeyPair]
+  );
+  const toggleSwitch = () => setIsActive((previousState) => !previousState);
```

Comment:
can this not be `setIsActive(!isActive)`?

Reply:
true, but i think its fine for now